firmware/src/widgets/weatherWidget.cpp

```diff
@@ -45,6 +45,7 @@ void WeatherWidget::setup() {
 }
 
 void WeatherWidget::draw(bool force) {
+    m_manager.setFont(DEFAULT_FONT);
```

Comment:
This does not seem like something the widget drawing needs to be responsible for? (setting the default font)

Reply:
It's more like the WeatherWidget saying that it wants to use the default font.

The selected font is global at the moment (see other comment), therefore each widget is responsible that it sets the font that it wants to use before calling drawString().

I would prefer having some kind of lifecycle like Android with onStart(), onPause(), onResume(), onStop().

Then we could put .setFont() in onResume().
We can't really put it in setup() because that's called for all widgets during boot. For me, setup() is like onStart() at the moment...